lib/Cake/Utility/Hash.php

```diff
@@ -843,10 +843,11 @@ public static function apply(array $data, $path, $function) {
  * @param string $path A Set-compatible path to the array value
  * @param string $dir See directions above. Defaults to 'asc'.
  * @param string $type See direction types above. Defaults to 'regular'.
+ * @param string $ignoreCase Case insensitive sorting. Defaults to false.
  * @return array Sorted array of data
  * @link http://book.cakephp.org/2.0/en/core-utility-libraries/hash.html#Hash::sort
  */
-	public static function sort(array $data, $path, $dir = 'asc', $type = 'regular') {
+	public static function sort(array $data, $path, $dir = 'asc', $type = 'regular', $ignoreCase = false) {
```

Comment:
I'm not a fan of boolean flags. They make method calls harder to understand at a glance. Could we either overload $type into an options array, or have more types?

Something like

```php
if (is_string($type)) {
  $type = ['type' => $type];
}
$type = (array)$type + ['type' => 'regular', 'caseSensitive' => true];
```
could be one way to overload the parameter.